lib/galaxy/tools/data/__init__.py

```diff
@@ -362,6 +366,11 @@ def configure_and_load(self, config_element, tool_data_path, from_shed_config=Fa
                     log.debug('Loading Data Table URL "%s" as filename "%s".', filename, tmp_file.name)
                     filename = tmp_file.name
                     tmp_file.flush()
+                else:
+                    # Pull the filename from a global config
+                    filename = file_element.get('from_config', None) or None
```

Comment:
```suggestion
                    filename = file_element.get('from_config', None)
```
It will be `None` regardless. Also, if `file_element` is a dictionary object, the second argument is not needed (same as comment below)

Reply:
the `or None` will make sure that `filename` remains `None` (as it was before it entered this logical scope) even if someone set the `from_config` to an empty string (an explicit empty value for the table config).

Reply:
Oh, OK. It'll still eval to `False` on the next line though ('' is falsy), but the extra safeguard doesn't hurt.